CRM/MembershipExtras/Hook/Post/EntityFinancialTrxn.php

```diff
@@ -41,13 +41,39 @@ public function updatePaymentPlanStatus() {
 
     $newStatus = $this->generatePaymentPlanNewStatus();
     if ($newStatus !== NULL) {
+      if ($newStatus === 'Completed') {
+        $subscriptionLines = $this->getSubscriptionLines();
+        foreach($subscriptionLines as $line) {
+          if (!empty($line['start_date']) && empty($line['end_date'])) {
+            civicrm_api3('ContributionRecurLineItem', 'create', [
+              'id' => $line['id'],
+              'end_date' => $this->recurContribution['end_date'],
+            ]);
+          }
+        }
+      }
+
       civicrm_api3('ContributionRecur', 'create', [
         'id' => $this->recurContribution['id'],
         'contribution_status_id' => $newStatus,
       ]);
     }
   }
 
+  /**
+   * Returns LineItems associated to a recurring contribution
+   * 
+   * @return array
+   */
+  private function getSubscriptionLines() {
+    $lines = civicrm_api3('ContributionRecurLineItem', 'get', [
+      'sequential' => 1,
+      'contribution' => $this->recurContribution['id'],
```

Comment:
civicrm API has a hardcoded limit to get only 25 result from any API call, so for example if there was 28 subscritopn lines then only 25 ones will be fetched from this API call, you can alter the API to fetch all the results by adding : 

```
'options' => ['limit' => 0],
```

like this : 

```
    $lines = civicrm_api3('ContributionRecurLineItem', 'get', [
      'sequential' => 1,
      'contribution' => $this->recurContribution['id'],
      'options' => ['limit' => 0],
    ]);
```